MINOR: Fix some java docs of ReplicaStateMachine

dengziming opened the pull request:
Modify the java docs of ReplicaStateMachine, the java docs is not in consistent with `OfflineReplica.validPreviousStates` and `OnlineReplica.validPreviousStates`

Move the  repeated `if(traceEnabled)` before `logSuccessfulTransition` method to the method

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

Commit:
Fix some inconsitent java docs

hachikuji commented on core/src/main/scala/kafka/controller/ReplicaStateMachine.scala:
```diff
@@ -423,9 +414,10 @@ class ZkReplicaStateMachine(config: KafkaConfig,
+  private def logSuccessfulTransition(traceEnabled: Boolean, logger: StateChangeLogger, replicaId: Int, partition: TopicPartition,
                                       currState: ReplicaState, targetState: ReplicaState): Unit = {
-    logger.trace(s"Changed state of replica $replicaId for partition $partition from $currState to $targetState")
+    if (traceEnabled)
```
Seems like we could just get rid of this. The call to `trace` will already check `isTraceEnabled` before building the message, so I'm not sure we're saving anything.

dengziming replied:
Done, also ping @stanislavkozlovski

hachikuji reviewed (commented):
Thanks, left a minor comment.

dengziming commented:
also ping @chia7712  to have a look

chia7712 commented:
> Seems like we could just get rid of this. The call to trace will already check isTraceEnabled before building the message, so I'm not sure we're saving anything.

The check was removed by https://github.com/apache/kafka/commit/ed8b0315a6c3705b2a163ce3ab4723234779264f#diff-d886593f0d0ebeee617016c3dfa5b5d2981d7a8f776bac939416615ba50a001eR51. We do build the message when calling ```msgWithLogIdent```. This is not related to this PR but we can add the check back to ```Logging#trace``` if we want to avoid checking ```isTraceEnabled``` from caller.

chia7712 commented:
@dengziming The docs fix LGTM. Maybe we can merge the code about docs fix. The issue about ```isTraceEnabled``` can be addressed in another PR. WDYT?

dengziming commented:
> @dengziming The docs fix LGTM. Maybe we can merge the code about docs fix. The issue about `isTraceEnabled` can be addressed in another PR. WDYT?
Thank you for your suggestions, I changed the code and also changed the title.

chia7712 reviewed (approved):
@dengziming Thanks for your patch. LGTM

chia7712 commented:
@dengziming Thanks for your patch. Feel free to open PR to handle ```isTraceEnabled```
